sys/vane/clay.hoon

```diff
@@ -3499,7 +3687,7 @@
       ^-  (map path lobe)
       ?:  =(0 yon)  ~
       ::  we use %z for the check because it looks at all child paths.
-      ?:  |(?=(~ for) (may-read u.for %z yon pax))  ~
+      ?.  |(?=(~ for) (may-read u.for %z yon pax))  ~
```

Comment:
@joemfb, I'm not sure, but I think this line was what caused the non-determinism of the `%child-sync` test before.  If stuff (setting permissions, starting the merge, etc) happened in a particular order, it triggered this bug so the child wouldn't get updates from its parent.